src/detect-cipservice.c

```diff
@@ -157,7 +157,11 @@ static DetectCipServiceData *DetectCipServiceParse(const char *rulestrc)
             goto error;
         }
 
-        sscanf(token, "%2" SCNu8, &var);
+        if (sscanf(token, "%2" SCNu8, &var) != 1) {
+            SCLogError("incorrectly formatted input");
```

Comment:
it would be good to add the incorrect output, and a hint on how what is acceptable

Reply:
The output for `sscanf` is supposed to be 1 since it's only takes & works on 1 input.

Any other output is incorrect.

Reply:
You mean, to add a comment to hint that `sscanf` is supposed to return the value of 1 and why it's so?

Reply:
sorry I meant the correct input, as a user isn't helped by "error!". The user needs to be told what the correct syntax is and what the incorrect input was. Otherwise we're just saying: "somewhere in this rule something went wrong, good luck"

Reply:
OK! I get the hang of what you mean,

Although, I'll need guidance on what to do; suggestions.

Reply:
Just for updates: This was addressed during a mentoring call.